octoprint_mrbeam/printing/comm_acc2.py

```diff
@@ -2564,27 +2570,52 @@ def cancelPrint(self, failed=False, error_msg=False):
     def setPause(
         self, pause, send_cmd=True, pause_for_cooling=False, trigger=None, force=False
     ):
-        if not self._currentFile:
-            return
+        """Pause or resume the print.
 
-        payload = self._get_printing_file_state()
-        payload["trigger"] = trigger
+        Args:
+            pause (bool): True to pause, False to resume.
+            send_cmd (bool): True to send a cycle_start realtime command to the printer.
+            pause_for_cooling (bool): True if the pause is for cooling, False otherwise.
+            trigger (str): The source of the pause request.
+            force (bool): True to force the pause, False otherwise.
 
+        Returns:
+            None
+        """
+        if not self._currentFile:
+            payload = {}
+        else:
+            payload = self._get_printing_file_state()
+        payload["trigger"] = trigger
+        self._logger.debug(
+            "pause %s send_cmd: %s trigger: %s force: %s ispaused: %s",
+            pause,
+            send_cmd,
+            trigger,
+            force,
+            self.isPaused(),
+        )
         if not pause and (self.isPaused() or force):
-            if self._pauseWaitStartTime:
-                self._pauseWaitTimeLost = self._pauseWaitTimeLost + (
-                    time.time() - self._pauseWaitStartTime
-                )
-                self._pauseWaitStartTime = None
-            self._pause_delay_time = time.time()
-            payload[
-                "time"
-            ] = self.getPrintTime()  # we need the pasue time to be removed from time
-            self.watch_dog.start()
             if send_cmd is True:
-                self._real_time_commands["cycle_start"] = True
-            self._send_event.set()
-            eventManager().fire(OctoPrintEvents.PRINT_RESUMED, payload)
+                self._real_time_commands[
+                    "cycle_start"
+                ] = True  # sends realtime command to resume print
+            if self.isPaused():
+                # continue from intentional pause
+                if self._pauseWaitStartTime:
+                    self._pauseWaitTimeLost = self._pauseWaitTimeLost + (
+                        time.time() - self._pauseWaitStartTime
+                    )
+                    self._pauseWaitStartTime = None
+                self._pause_delay_time = time.time()  # get_uptime()
```

Comment:
Are we planning to use get_uptime() in the future, or what is this comment about?